[visionOS] Captions are missing in fullscreen mode

jernoble opened the pull request:
#### a8256bf61cd70cb214e683a81a617354b57b3a42
<pre>
[visionOS] Captions are missing in fullscreen mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=274592">https://bugs.webkit.org/show_bug.cgi?id=274592</a>
<a href="https://rdar.apple.com/128502343">rdar://128502343</a>

Reviewed by Andy Estes.

When enabling the entity-backed rendering path on visionOS, we inadventantly broke
caption display in fullscreen. The reasons are multifaceted, but stem from the lack
of a WebAVPlayerLayer containing the video content in that path.

Untangling the requirement to have a fullscreen video layer requires changes at every
process level:

- Rather than have individual MediaPlayers report whether they need a TextTrackRepresentation,
  move responibility for telling HTMLMediaPlayer so up to the UI layer (and VideoPlaybackInterface).
- Notify the HTMLMediaElement of TextTrackRepresentation size changes directly, rather than
  implicitly via fullscreen video layer size changes.
- Allow the bounds of the TextTrackRepresentation to be set directly, rather than implictly
  via the representation&apos;s layer.

This removes large swaths of text track responsibilities from MediaPlayer/Private, but requires
VideoPresentationInterface subclasess to directly notify the element when they need a
representation for text tracks to be created.

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::textTrackRepresentation const):
(WebCore::MediaControlsHost::requiresTextTrackRepresentationChanged):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setTextTrackRepresentataionBounds):
(WebCore::HTMLMediaElement::setRequiresTextTrackRepresentation):
(WebCore::HTMLMediaElement::requiresTextTrackRepresentation const):
(WebCore::HTMLMediaElement::setTextTrackRepresentation):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::requiresTextTrackRepresentationChanged):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.h:
* Source/WebCore/platform/cocoa/VideoFullscreenCaptions.h:
* Source/WebCore/platform/cocoa/VideoFullscreenCaptions.mm:
(WebCore::VideoFullscreenCaptions::setTrackRepresentationImage):
* Source/WebCore/platform/cocoa/VideoPresentationModel.h:
(WebCore::VideoPresentationModel::setRequiresTextTrackRepresentation):
(WebCore::VideoPresentationModel::setTextTrackRepresentationBounds):
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.h:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
(WebCore::VideoPresentationModelVideoElement::setVideoFullscreenFrame):
(WebCore::VideoPresentationModelVideoElement::setRequiresTextTrackRepresentation):
(WebCore::VideoPresentationModelVideoElement::setTextTrackRepresentationBounds):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/MediaPlayerPrivate.h:
(WebCore::MediaPlayerPrivateInterface::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/TextTrackRepresentation.cpp:
* Source/WebCore/platform/graphics/TextTrackRepresentation.h:
* Source/WebCore/platform/graphics/VideoLayerManager.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/VideoLayerManagerObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoLayerManagerObjC.mm:
(WebCore::VideoLayerManagerObjC::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.h:
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
(WebCore::MediaPlayerPrivateWebM::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/cocoa/TextTrackRepresentationCocoa.h:
* Source/WebCore/platform/graphics/cocoa/TextTrackRepresentationCocoa.mm:
(WebCore::TextTrackRepresentationCocoa::setBounds):
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.mm:
(WebCore::VideoPresentationInterfaceIOS::willStartPictureInPicture):
(WebCore::VideoPresentationInterfaceIOS::finalizeSetup):
(WebCore::VideoPresentationInterfaceIOS::setMode):
(WebCore::VideoPresentationInterfaceIOS::clearMode):
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
* Source/WebCore/platform/mac/VideoPresentationInterfaceMac.mm:
(-[WebVideoPresentationInterfaceMacObjC pipDidClose:]):
(WebCore::VideoPresentationInterfaceMac::setMode):
(WebCore::VideoPresentationInterfaceMac::clearMode):
(WebCore::VideoPresentationInterfaceMac::setupFullscreen):
* Source/WebKit/Platform/ios/PlaybackSessionInterfaceLMK.h:
* Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.h:
* Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm:
(-[WKLinearMediaKitCaptionsLayer initWithParent:]):
(-[WKLinearMediaKitCaptionsLayer layoutSublayers]):
(WebKit::VideoPresentationInterfaceLMK::captionsLayer):
(WebKit::VideoPresentationInterfaceLMK::captionsLayerBoundsChanged):
* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
(WebKit::VideoPresentationModelContext::setVideoFullscreenFrame):
(WebKit::VideoPresentationModelContext::setRequiresTextTrackRepresentation):
(WebKit::VideoPresentationModelContext::setTextTrackRepresentationBounds):
(WebKit::VideoPresentationManagerProxy::setRequiresTextTrackRepresentation):
(WebKit::VideoPresentationManagerProxy::setTextTrackRepresentationBounds):
(WebKit::VideoPresentationManagerProxy::setVideoFullscreenFrame):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::requiresTextTrackRepresentation const): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.h:
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.mm:
(WebKit::WebTextTrackRepresentationCocoa::setBounds):
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.h:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.messages.in:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.mm:
(WebKit::VideoPresentationManager::setVideoFullscreenFrame):
(WebKit::VideoPresentationManager::setRequiresTextTrackRepresentation):
(WebKit::VideoPresentationManager::setTextTrackRepresentationBounds):

Canonical link: <a href="https://commits.webkit.org/279405@main">https://commits.webkit.org/279405@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0ce2948135c1588d3a72c1697a0f6267076201c7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53073 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32410 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/5560 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56352 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/3796 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55378 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/39274 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/3523 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43037 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2451 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55171 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/30168 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/45811 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24165 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/27176 "Found unexpected failure with change (failure)") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3133 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/1955 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/49587 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3291 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/57947 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28214 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/3247 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/50444 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/29434 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46031 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/49755 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11639 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/30353 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29188 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
```diff
@@ -225,6 +225,13 @@
+{
+    INFO_LOG_IF_POSSIBLE(LOGIDENTIFIER, rect.size());
+    if (m_videoElement)
+        m_videoElement->setVideoFullscreenFrame(rect);
```
```suggestion
    if (RefPtr videoElement = m_videoElement)
        videoElement->setVideoFullscreenFrame(rect);
```

aestes commented on Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
```diff
@@ -341,6 +348,24 @@
+        return;
+
+    ALWAYS_LOG_IF_POSSIBLE(LOGIDENTIFIER);
+    m_videoElement->setRequiresTextTrackRepresentation(requiresTextTrackRepresentation);
```
```suggestion
    RefPtr videoElement = m_videoElement;
    if (!videoElement)
        return;

    ALWAYS_LOG_IF_POSSIBLE(LOGIDENTIFIER);
    videoElement->setRequiresTextTrackRepresentation(requiresTextTrackRepresentation);
```

aestes commented on Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
```diff
@@ -341,6 +348,24 @@
+        return;
+
+    ALWAYS_LOG_IF_POSSIBLE(LOGIDENTIFIER, bounds.size());
+    m_videoElement->setTextTrackRepresentataionBounds(bounds);
```
```suggestion
    RefPtr videoElement = m_videoElement;
    if (!videoElement)
        return;

    ALWAYS_LOG_IF_POSSIBLE(LOGIDENTIFIER, bounds.size());
    videoElement->setTextTrackRepresentataionBounds(bounds);
```

aestes commented on Source/WebCore/platform/mac/VideoPresentationInterfaceMac.mm:
```diff
@@ -409,10 +410,14 @@ - (void)pipActionStop:(PIPViewController *)pip
 
     m_mode = newMode;
 
+    auto model = videoPresentationModel();
```
Nit: maybe replace `auto` with `RefPtr` to make it clear that `model` is retained.

aestes commented on Source/WebCore/platform/mac/VideoPresentationInterfaceMac.mm:
```diff
@@ -424,10 +429,14 @@ - (void)pipActionStop:(PIPViewController *)pip
 
     m_mode = newMode;
 
+    auto model = videoPresentationModel();
```
Ditto

aestes commented on Source/WebCore/platform/mac/VideoPresentationInterfaceMac.mm:
```diff
@@ -461,8 +470,10 @@ - (void)pipActionStop:(PIPViewController *)pip
 
     RunLoop::main().dispatch([protectedThis = Ref { *this }, this] {
-        if (auto model = videoPresentationModel())
+        if (auto model = videoPresentationModel()) {
```
Ditto

aestes commented on Source/WebKit/Platform/ios/PlaybackSessionInterfaceLMK.h:
```diff
@@ -35,6 +35,10 @@ OBJC_CLASS WKLinearMediaPlayerDelegate;
 
+class PlaybackSessionInterfaceLMKClient {
+
+};
```
Is this needed?

jernoble replied:
Not for this patch!

aestes commented on Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm:
```diff
@@ -36,21 +36,33 @@
+- (void)layoutSublayers
+{
+    if (RefPtr parent = _parent.get())
+        parent->captionsLayerBoundsChanged(self.bounds);
```
Nit: this is probably fine, but I sorta feel like this should come after `[super layoutSublayers]` so that this layer's hierarchy is in a consistent state.

aestes commented on Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm:
```diff
@@ -161,6 +173,15 @@ - (void)layoutSublayers
+#if HAVE(SPATIAL_TRACKING_LABEL)
+    [m_spatialTrackingLayer setPosition:bounds.center()];
+#endif
+    if (auto model = videoPresentationModel())
```
Ditto (re: `RefPtr` instead of `auto`)

aestes commented on Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
```diff
@@ -446,6 +453,18 @@ - (BOOL)prefersStatusBarHidden
+void VideoPresentationModelContext::setRequiresTextTrackRepresentation(bool requiresTextTrackRepresentation)
+{
+    if (m_manager)
+        m_manager->setRequiresTextTrackRepresentation(m_contextId, requiresTextTrackRepresentation);
```
```suggestion
    if (RefPtr manager = m_manager.get())
        manager->setRequiresTextTrackRepresentation(m_contextId, requiresTextTrackRepresentation);
```

aestes commented on Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
```diff
@@ -446,6 +453,18 @@ - (BOOL)prefersStatusBarHidden
+void VideoPresentationModelContext::setTextTrackRepresentationBounds(const IntRect& bounds)
+{
+    if (m_manager)
+        m_manager->setTextTrackRepresentationBounds(m_contextId, bounds);
```
```suggestion
    if (RefPtr manager = m_manager.get())
        manager->setTextTrackRepresentationBounds(m_contextId, bounds);
```

aestes commented on Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
```diff
@@ -1105,6 +1124,18 @@ - (BOOL)prefersStatusBarHidden
+void VideoPresentationManagerProxy::setRequiresTextTrackRepresentation(PlaybackSessionContextIdentifier contextId , bool requiresTextTrackRepresentation)
+{
+    if (m_page)
+        m_page->send(Messages::VideoPresentationManager::SetRequiresTextTrackRepresentation(contextId, requiresTextTrackRepresentation));
```
```suggestion
    if (RefPtr page = m_page.get())
        page->send(Messages::VideoPresentationManager::SetRequiresTextTrackRepresentation(contextId, requiresTextTrackRepresentation));
```

aestes commented on Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
```diff
@@ -1105,6 +1124,18 @@ - (BOOL)prefersStatusBarHidden
+void VideoPresentationManagerProxy::setTextTrackRepresentationBounds(PlaybackSessionContextIdentifier contextId , const IntRect& bounds)
+{
+    if (m_page)
+        m_page->send(Messages::VideoPresentationManager::SetTextTrackRepresentationBounds(contextId, bounds));
```
```suggestion
    if (RefPtr page = m_page.get())
        page->send(Messages::VideoPresentationManager::SetTextTrackRepresentationBounds(contextId, bounds));
```

aestes commented on Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
```diff
@@ -1316,6 +1347,12 @@ - (BOOL)prefersStatusBarHidden
+void VideoPresentationManagerProxy::setVideoFullscreenFrame(PlaybackSessionContextIdentifier contextId, WebCore::FloatRect frame)
+{
+    if (m_page)
+        m_page->send(Messages::VideoPresentationManager::SetVideoFullscreenFrame(contextId, frame));
```
```suggestion
    if (RefPtr page = m_page.get())
        page->send(Messages::VideoPresentationManager::SetVideoFullscreenFrame(contextId, frame));
```

jernoble committed:
[visionOS] Captions are missing in fullscreen mode
https://bugs.webkit.org/show_bug.cgi?id=274592
rdar://128502343

Reviewed by Andy Estes.

When enabling the entity-backed rendering path on visionOS, we inadventantly broke
caption display in fullscreen. The reasons are multifaceted, but stem from the lack
of a WebAVPlayerLayer containing the video content in that path.

Untangling the requirement to have a fullscreen video layer requires changes at every
process level:

- Rather than have individual MediaPlayers report whether they need a TextTrackRepresentation,
  move responibility for telling HTMLMediaPlayer so up to the UI layer (and VideoPlaybackInterface).
- Notify the HTMLMediaElement of TextTrackRepresentation size changes directly, rather than
  implicitly via fullscreen video layer size changes.
- Allow the bounds of the TextTrackRepresentation to be set directly, rather than implictly
  via the representation's layer.

This removes large swaths of text track responsibilities from MediaPlayer/Private, but requires
VideoPresentationInterface subclasess to directly notify the element when they need a
representation for text tracks to be created.

* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::textTrackRepresentation const):
(WebCore::MediaControlsHost::requiresTextTrackRepresentationChanged):
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::setTextTrackRepresentataionBounds):
(WebCore::HTMLMediaElement::setRequiresTextTrackRepresentation):
(WebCore::HTMLMediaElement::requiresTextTrackRepresentation const):
(WebCore::HTMLMediaElement::setTextTrackRepresentation):
* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.cpp:
(WebCore::MediaControlTextTrackContainerElement::requiresTextTrackRepresentationChanged):
* Source/WebCore/html/shadow/MediaControlTextTrackContainerElement.h:
* Source/WebCore/platform/cocoa/VideoFullscreenCaptions.h:
* Source/WebCore/platform/cocoa/VideoFullscreenCaptions.mm:
(WebCore::VideoFullscreenCaptions::setTrackRepresentationImage):
* Source/WebCore/platform/cocoa/VideoPresentationModel.h:
(WebCore::VideoPresentationModel::setRequiresTextTrackRepresentation):
(WebCore::VideoPresentationModel::setTextTrackRepresentationBounds):
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.h:
* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
(WebCore::VideoPresentationModelVideoElement::setVideoFullscreenFrame):
(WebCore::VideoPresentationModelVideoElement::setRequiresTextTrackRepresentation):
(WebCore::VideoPresentationModelVideoElement::setTextTrackRepresentationBounds):
* Source/WebCore/platform/graphics/MediaPlayer.cpp:
(WebCore::MediaPlayer::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/MediaPlayerPrivate.h:
(WebCore::MediaPlayerPrivateInterface::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/TextTrackRepresentation.cpp:
* Source/WebCore/platform/graphics/TextTrackRepresentation.h:
* Source/WebCore/platform/graphics/VideoLayerManager.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateAVFoundationObjC.mm:
(WebCore::MediaPlayerPrivateAVFoundationObjC::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::MediaPlayerPrivateMediaSourceAVFObjC::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/VideoLayerManagerObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/VideoLayerManagerObjC.mm:
(WebCore::VideoLayerManagerObjC::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.h:
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
(WebCore::MediaPlayerPrivateWebM::requiresTextTrackRepresentation const): Deleted.
* Source/WebCore/platform/graphics/cocoa/TextTrackRepresentationCocoa.h:
* Source/WebCore/platform/graphics/cocoa/TextTrackRepresentationCocoa.mm:
(WebCore::TextTrackRepresentationCocoa::setBounds):
* Source/WebCore/platform/ios/VideoPresentationInterfaceIOS.mm:
(WebCore::VideoPresentationInterfaceIOS::willStartPictureInPicture):
(WebCore::VideoPresentationInterfaceIOS::finalizeSetup):
(WebCore::VideoPresentationInterfaceIOS::setMode):
(WebCore::VideoPresentationInterfaceIOS::clearMode):
* Source/WebCore/platform/ios/WebVideoFullscreenControllerAVKit.mm:
* Source/WebCore/platform/mac/VideoPresentationInterfaceMac.mm:
(-[WebVideoPresentationInterfaceMacObjC pipDidClose:]):
(WebCore::VideoPresentationInterfaceMac::setMode):
(WebCore::VideoPresentationInterfaceMac::clearMode):
(WebCore::VideoPresentationInterfaceMac::setupFullscreen):
* Source/WebKit/Platform/ios/PlaybackSessionInterfaceLMK.h:
* Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.h:
* Source/WebKit/Platform/ios/VideoPresentationInterfaceLMK.mm:
(-[WKLinearMediaKitCaptionsLayer initWithParent:]):
(-[WKLinearMediaKitCaptionsLayer layoutSublayers]):
(WebKit::VideoPresentationInterfaceLMK::captionsLayer):
(WebKit::VideoPresentationInterfaceLMK::captionsLayerBoundsChanged):
* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.h:
* Source/WebKit/UIProcess/Cocoa/VideoPresentationManagerProxy.mm:
(WebKit::VideoPresentationModelContext::setVideoFullscreenFrame):
(WebKit::VideoPresentationModelContext::setRequiresTextTrackRepresentation):
(WebKit::VideoPresentationModelContext::setTextTrackRepresentationBounds):
(WebKit::VideoPresentationManagerProxy::setRequiresTextTrackRepresentation):
(WebKit::VideoPresentationManagerProxy::setTextTrackRepresentationBounds):
(WebKit::VideoPresentationManagerProxy::setVideoFullscreenFrame):
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp:
(WebKit::MediaPlayerPrivateRemote::requiresTextTrackRepresentation const): Deleted.
* Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.h:
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.h:
* Source/WebKit/WebProcess/cocoa/TextTrackRepresentationCocoa.mm:
(WebKit::WebTextTrackRepresentationCocoa::setBounds):
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.h:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.messages.in:
* Source/WebKit/WebProcess/cocoa/VideoPresentationManager.mm:
(WebKit::VideoPresentationManager::setVideoFullscreenFrame):
(WebKit::VideoPresentationManager::setRequiresTextTrackRepresentation):
(WebKit::VideoPresentationManager::setTextTrackRepresentationBounds):

Canonical link: https://commits.webkit.org/279405@main
